rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/range/20_synthetic_source.yml

```diff
@@ -0,0 +1,473 @@
+
+  - do:
+      index:
+        index: synthetic_source_test
+        id: "6"
+        body: { "double_range": { "gte": 1 } }
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      search:
+        index: synthetic_source_test
+  - match: { hits.total.value: 6 }
+  - match:
+      hits.hits.0._source:
+        double_range:  { "gte": 1.0, "lte": 5.0 }
+  - match:
+      hits.hits.1._source:
+        double_range: { "gte": 4.0, "lte": 5.0 }
+  - match:
+      hits.hits.2._source:
+        double_range: [ { "gte": 4.0, "lte": 7.0 }, { "gte": 4.0, "lte": 8.0 } ]
+  - match:
+      hits.hits.3._source: {}
+  - match:
+      hits.hits.4._source:
+        double_range: { "gte": "-Infinity", "lte": 10.0 }
+  - match:
+      hits.hits.5._source:
+        double_range: { "gte": 1.0, "lte": "Infinity" }
+
+---
+"IP range":
+
+  - do:
+      index:
+        index: synthetic_source_test
+        id: "1"
+        body: { "ip_range" : { "gte": "192.168.0.1", "lte": "192.168.0.5" } }
+
+  - do:
+      index:
+        index: synthetic_source_test
+        id: "2"
+        body: { "ip_range" : { "gt": "192.168.0.1", "lte": "192.168.0.3" } }
+
+  - do:
+      index:
+        index: synthetic_source_test
+        id: "3"
+        body: { "ip_range" : [ { "gte": "192.168.0.4", "lt": "192.168.0.5" } ] }
+
+  - do:
+      index:
+        index: synthetic_source_test
+        id: "4"
+        body: { "ip_range" : { "gt": "192.168.0.4", "lt": "192.168.0.8" } }
+
+  - do:
+      index:
+        index: synthetic_source_test
+        id: "5"
+        body: { "ip_range" : "74.125.227.0/25" }
+
+  - do:
+      index:
+        index: synthetic_source_test
+        id: "6"
+        body: { "ip_range" : null }
+
+  - do:
+      index:
+        index: synthetic_source_test
+        id: "7"
+        body: { "ip_range": { "gte": null, "lte": "10.10.10.10" } }
+
+  - do:
+      index:
+        index: synthetic_source_test
+        id: "8"
+        body: { "ip_range": { "gte": "10.10.10.10" } }
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      search:
+        index: synthetic_source_test
+  - match: { hits.total.value: 8 }
+  - match:
+      hits.hits.0._source:
+        ip_range:  { "gte": "192.168.0.1", "lte": "192.168.0.5" }
+  - match:
+      hits.hits.1._source:
+        ip_range: { "gte": "192.168.0.2", "lte": "192.168.0.3" }
+  - match:
+      hits.hits.2._source:
+        ip_range: { "gte": "192.168.0.4", "lte": "192.168.0.4" }
+  - match:
+      hits.hits.3._source:
+        ip_range: { "gte": "192.168.0.5", "lte": "192.168.0.7" }
+  - match:
+      hits.hits.4._source:
+        ip_range: { "gte": "74.125.227.0", "lte": "74.125.227.127" }
+  - match:
+      hits.hits.5._source: {}
+  - match:
+      hits.hits.6._source:
+        ip_range: { "gte": "::", "lte": "10.10.10.10" }
+  - match:
+      hits.hits.7._source:
+        ip_range: { "gte": "10.10.10.10", "lte": "ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff" }
```

Comment:
This mix of IPv6 and IPv4 is pretty unfortunate but i don't see a good way to address that. We could look at other end of range and try to format it in the same way but it's pretty brittle in my opinion.

Reply:
Actually our docs say that we support mixed ranges so maybe this is okay?

Reply:
Yes that would be ideal, let me see if i can do that.